sdk/rust/src/key_value.rs

```diff
@@ -61,6 +64,31 @@ impl Store {
     pub fn get_keys(&self) -> Result<Vec<String>, Error> {
         key_value::get_keys(self.0)
     }
+
+    #[cfg(feature = "serde")]
+    /// Serialize the given data to JSON, then set it as the value for the specified key.
+    pub fn set_json<T>(&self, key: impl AsRef<str>, value: &T) -> Result<(), anyhow::Error>
+    where
+        T: ?Sized + Serialize,
+    {
+        Ok(key_value::set(
+            self.0,
+            key.as_ref(),
+            &serde_json::to_vec(value)?,
+        )?)
+    }
+
+    #[cfg(feature = "serde")]
+    /// Deserialize an instance of type `T` from the value of `key`.
+    pub fn get_json<T>(&self, key: impl AsRef<str>) -> Result<T, anyhow::Error>
```

Comment:
For naming I think `{get,set}_json` is fine. I could imagine `{get,set}_as`, but that hides the fact that we assume json...